### DISCUSSION TITLE
Integration in .NET Maui

### DISCUSSION THREAD

**Discussion by dnx231** [TolikPylypchuk/SharpHook#17] [Q&A] [answered]
Hello,
I am trying to get SharpHook running under .NET Maui.
After starting the project on Android Emulator and Android Device, I get the following error message:

`[monodroid-assembly] Shared library 'uiohook' not loaded, p/invoke 'hook_set_dispatch_proc' may fail
[monodroid-assembly] Shared library 'uiohook' not loaded, p/invoke 'hook_run' may fail
[monodroid-assembly] Shared library 'uiohook' not loaded, p/invoke 'hook_set_dispatch_proc' may fail
**SharpHook.HookException:** 'Hook exception based on result: Failure'
`

Does anyone have any idea what this could be? Or has anyone ever tried to use SharpHook under .NET Maui?

Regards

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for using SharpHook! Unfortunately, it just doesn't work on Android and iOS. libuiohook - the library which SharpHook wraps - only works on desktop OSes: Windows, macOS, and Linux. I'm not sure there even is intention to support Android and iOS. So, SharpHook should work with MAUI, but only on Windows and macOS.

**Reply by dnx231** [reply]
Hi Tolik,
Thanks for the quick feedback.

Damn, I would have better read the specifications carefully. ;-) My fault...

**Reply by TolikPylypchuk** [reply]
No worries, I'm here to answer any questions :)
